tests/e2e-tests/test-devicestate.sh

```diff
@@ -0,0 +1,98 @@
+# Insert file header
+
+# Insert function header
+function cleanup() {
+	echo Starting cleanup
+	
+	local armToken=$1
+	local apiToken=$2
+	local device_id=$3
+	local token_id=$4
+
+	# Clean up device if it exists
+	local device_exists=$(curl -X GET -H "Authorization:$apiToken" https://e2etest-iotc-iiot-asset-app.azureiotcentral.com/api/preview/devices/${device_id}  2>/dev/null | jq -r '.id')	
+	if [ "$device_exists" == "$device_id" ];
+	then
+		echo Device ${device_id} exists
+		echo Delete device ${device_id}
+		curl -X DELETE -H "Authorization:Bearer $armToken" https://e2etest-iotc-iiot-asset-app.azureiotcentral.com/api/preview/devices/${device_id} 2>/dev/null
+	else
+		echo Device ${device_id} does not exist
+	fi;
+
+	# Clean up API token if it exists
+	local apiToken_exists=$(curl -X GET -H "Authorization:$apiToken" https://e2etest-iotc-iiot-asset-app.azureiotcentral.com/api/preview/apiTokens/${token_id} 2>/dev/null | jq -r '.id')
+	if [ "$apiToken_exists" == "$token_id" ];
+	then
+		echo API token ${token_id} exists
+		echo Delete API token ${token_id}
+		curl -X DELETE -H "Authorization:Bearer $armToken" https://e2etest-iotc-iiot-asset-app.azureiotcentral.com/api/preview/apiTokens/${token_id} 2>/dev/null
+	else
+		echo API token ${token_id} does not exist
+	fi;
+
+	echo Completed cleanup
+}
+
+# Azure IoT DDE team subscription
+subscription=377c3343-75bb-4244-98a3-0fb84a830c4b
+
+# Create a random number to make sure create new resources per each run
+let num=$RANDOM*$RANDOM
+prefix=e2e
+token_id="${prefix}testtoken${num}"
+device_id="${prefix}testdevice${num}"
+device_displayName=$device_id
+device_template=dtmi:z3kvj66agb:gi4gydpmx 
+test_result=1 # fail by default
+
+# The Central app is stored in PipelineResources-IoTEdgeConfig resource group
+echo Get access token to subscription "Azure IoT DDE team subscription"
+armToken=$(az account get-access-token --resource https://apps.azureiotcentral.com -s ${subscription} | jq -r '.accessToken')
```

Comment:
I didn't have jq installed when I ran this script for the first time, we might need to install this dependency before running the script

Reply:
the pipeline will do this step